azurelinuxagent/ga/monitor.py

```diff
@@ -193,3 +193,10 @@ def daemon(self):
             except Exception as e:
                 logger.warn("Failed to send events: {0}", e)
             time.sleep(60)
+
+    def add_sysinfo(self, event):
+        sysinfo_names = [v.name for v in self.sysinfo]
+        for param in event.parameters:
+            if param.name in sysinfo_names:
+                event.parameters.remove(param)
```

Comment:
we might want some debug logging here, to follow up with extension owners at some point.